Kernel/FileSystem/FATFS/Inode.cpp

```diff
@@ -228,6 +252,59 @@ ErrorOr<void> FATInode::remove_last_cluster_from_chain()
     return {};
 }
 
+ErrorOr<Vector<FATInode::EntryLocation>> FATInode::allocate_entries(u32 count)
+{
+    // FIXME: This function ignores unused entries, we should make use of them
+
+    auto blocks = TRY(read_block_list());
+    auto entries = reinterpret_cast<FATEntry*>(blocks->data());
+
+    auto const entries_per_block = fs().logical_block_size() / sizeof(FATEntry);
+
+    Vector<EntryLocation> locations;
+
+    for (u32 current_entry_index = 0; current_entry_index < blocks->size() / sizeof(FATEntry); current_entry_index++) {
+        auto& entry = entries[current_entry_index];
+        if (entry.filename[0] == end_entry_byte) {
+            while (current_entry_index < blocks->size() / sizeof(FATEntry) && locations.size() < count) {
+                u32 chosen_block_index = current_entry_index / entries_per_block;
+                u32 chosen_entry_index = current_entry_index % entries_per_block;
+                locations.append({ m_block_list[chosen_block_index], chosen_entry_index });
```

Comment:
If the above is true then `unchecked_append` and if not, can we get away with `try_append` or do we need some cleanup when exiting from here